### PR TITLE
IGNITE-22655: Sql. incorrect error for quoted SQL

### PR DESCRIPTION
**Pull request by lowka**
Adds an additional SqlShuttle to validate identifiers.

https://issues.apache.org/jira/browse/IGNITE-22655

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlUpdate.java [hunk 1]
```diff
@@ -37,7 +38,7 @@
 public class IgniteSqlUpdate extends SqlUpdate {
 
     /** UPDATE operator. */
-    protected static final class Operator extends IgniteSqlSpecialOperator {
+    protected static final class Operator extends SqlSpecialOperator {
```
Comment (lowka) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlUpdate.java:41]: This was a typo. UPDATE is not a special operator at all.

Comment (ygerzhedovich) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlUpdate.java:41]: the comment is funny, due to it is still Special operator

File: modules/sql-engine/src/integrationTest/sql/identifiers/test_long_identifiers.test [hunk 2]
```diff
@@ -217,7 +221,7 @@ ALTER ZONE zoneName_veryLooooooooooooooooooooooooooooooooooooooooooooooooooooooo
 statement ok
 ALTER ZONE zoneName_veryLooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooongIdentifierOf128Characters RENAME TO zoneName;
 
-statement error: Failed to parse query: Length of identifier 
+statement error: Length of identifier
 ALTER ZONE zoneName RENAME TO zoneName_veryLoooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooongIdentifierOf129Characters;
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/identifiers/test_long_identifiers.test:226]: try to add a new test:
```
statement error: Length of identifier
SELECT * FROM t where 1 = (SELECT 1 from t as columnAlias_veryLooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooongIdentifierOf129Characters LIMIT 1);
```

Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/identifiers/test_long_identifiers.test:232]: Fixed.

**Review by ygerzhedovich — APPROVED** [review]

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java [hunk 3]
```diff
@@ -55,6 +55,8 @@ public final class IgniteSqlParser  {
     public static final SqlParser.Config PARSER_CONFIG = SqlParser.config()
             .withParserFactory(InternalIgniteSqlParser.FACTORY)
             .withLex(Lex.ORACLE)
+            // Do not validate the length of SQL identifiers.
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java:58]: it'a better to provide reason why do we disable validation here

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java:1]: Why don't we just introduced simple visitor to find all SqlIdentifier's in the tree and validate them?

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java [hunk 4]
```diff
@@ -192,6 +204,8 @@ private static SqlException convertException(SqlParseException ex) {
                 messageFromCause = messageFromCause.substring(0, messageFromCause.length() - 1);
             }
 
+            ex.printStackTrace(System.err);
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java]: Thanks. Fixed.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java:1]: Rewrote using a visitor.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java (hunk 3 above)
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java]: Fixed.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java [hunk 5]
```diff
@@ -273,4 +290,50 @@ private static SqlNode fixNodesIfNecessary(SqlNode node) {
...
+            }
+
+            return super.visit(id);
+        }
+
+        @Override
+        public @Nullable SqlNode visit(SqlCall call) {
+            SqlOperator operator = call.getOperator();
+
+            // If something when wrong during the parsing, fail at the validation stage
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParser.java:308]:
```suggestion
            // If something went wrong during the parsing, fail at the validation stage
```

**Review by korlov42 — APPROVED** [review]

**Pull request merged**